prometheus/promhttp/http.go

```diff
@@ -381,6 +387,14 @@ type HandlerOpts struct {
 	ProcessStartTime time.Time
 }
 
+type OpenMetricsOptions struct {
+	Enable bool
```

Comment:
Perhaps we could remove this field and use `OpenMetricsOptions *OpenMetricsOptions` to ensure it's optional? We cannot remove `EnableOpenMetrics` anyway so we could still use that, OR when OpenMetricsOptions is allocated assume OM is enabled? 🤔

Reply:
[Here](https://github.com/prometheus/client_golang/pull/1392/files#diff-8dc7897d4d9d5bf1c91a5fe1c931db556eabc1a6458dfce5ddd087b0bc5eec78R163-L167) I was kinda tying the possibility of enabling the optional unit to OM been already negotiated as the output format, but maybe there are better ways to ensure that.